From ad2ac0ad679550a7899729993eeb2524f5f94c9c Mon Sep 17 00:00:00 2001
Message-Id: <ad2ac0ad679550a7899729993eeb2524f5f94c9c.1350045285.git.minovotn@redhat.com>
In-Reply-To: <8d140e6fe679a73fd193609be44d508f17f7892b.1350045285.git.minovotn@redhat.com>
References: <8d140e6fe679a73fd193609be44d508f17f7892b.1350045285.git.minovotn@redhat.com>
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Thu, 11 Oct 2012 18:24:22 +0200
Subject: [PATCH 10/12] blockdev: do_change_block(): pass error message to
 OpenFileFailed error

RH-Author: Luiz Capitulino <lcapitulino@redhat.com>
Message-id: <1349979864-509-5-git-send-email-lcapitulino@redhat.com>
Patchwork-id: 42993
O-Subject: [RHEL6.4 qemu-kvm PATCH 4/6] blockdev: do_change_block(): pass error message to OpenFileFailed error
Bugzilla: 806775
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Pavel Hrdina <phrdina@redhat.com>

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
---
 blockdev.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 blockdev.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/blockdev.c b/blockdev.c
index 4b705de..052a962 100644
--- a/blockdev.c
+++ b/blockdev.c
@@ -1147,6 +1147,7 @@ int do_change_block(Monitor *mon, const char *device,
     BlockDriverState *bs;
     BlockDriver *drv = NULL;
     int bdrv_flags;
+    int ret;
 
     bs = bdrv_find(device);
     if (!bs) {
@@ -1165,8 +1166,9 @@ int do_change_block(Monitor *mon, const char *device,
     }
     bdrv_flags = bdrv_get_type_hint(bs) == BDRV_TYPE_CDROM ? 0 : BDRV_O_RDWR;
     bdrv_flags |= bdrv_is_snapshot(bs) ? BDRV_O_SNAPSHOT : 0;
-    if (bdrv_open(bs, filename, bdrv_flags, drv)) {
-        qerror_report(QERR_OPEN_FILE_FAILED, filename, "");
+    ret = bdrv_open(bs, filename, bdrv_flags, drv);
+    if (ret < 0) {
+        qerror_report(QERR_OPEN_FILE_FAILED, filename, strerror(-ret));
         return -1;
     }
     return monitor_read_bdrv_key_start(mon, bs, NULL, NULL);
-- 
1.7.11.7

